### PR TITLE
Bug 1499857 - remove support for claimedCredentials

### PR DESCRIPTION
**Pull request by djmitche**
The taskcluster-pulse service is being removed, so this option not
longer makes sense.

### CODE REVIEW

File: src/client.js [hunk 1]
```diff
@@ -217,11 +217,10 @@ class Client extends events.EventEmitter {
   }
 
   /**
-   * Update the _recycleTimer, either to expire at recycleAt or, if that's omitted,
-   * after recycleInterval.
+   * Update the _recycleTimer based on recycleInterval
    */
-  async _updateRecycleTimer(recycleAt) {
-    const recycleAfter = recycleAt ?  recycleAt - new Date() : this._recycleInterval;
+  async _updateRecycleTimer() {
```
Comment (Biboswan) [inline src/client.js]: This function was neccessary to deal with recycleAfter from tc-pulse. So I think this function now unnecessarily complicates logic unless there is some compatibility issue.

Comment (djmitche) [reply src/client.js]: We need some logic to handle this timer, and this seems like a good place for it.  What alternative would you suggest?

Comment (Biboswan) [reply src/client.js]: Use `setInterval(this.recycle, this._recycleInterval)` in the `Client` constructor as before.

**Comment by djmitche** [comment]
Updated to use `setInterval`.

**Review by Biboswan — APPROVED** [review]

**Comment by djmitche** [comment]
released as v10.0.0 (it's a breaking change, and this library should be numbered >10 just like all of the other taskcluster-lib-*)

**Pull request merged**